src/components/DataRenderFunc.js

```diff
@@ -0,0 +1,236 @@
+import React, {useState, useEffect} from "react";
+import "./DataRenderStyles.css";
+import Button from 'react-bootstrap/Button';
+import { Row } from "simple-flexbox";
+import "bootstrap/dist/css/bootstrap.min.css";
+import {
+    AreaChart,
+    Area,
+    LineChart,
+    Line,
+    XAxis,
+    YAxis,
+    CartesianGrid,
+    Tooltip,
+    Legend,
+    ResponsiveContainer,
+    LabelList
+} from "recharts";
+import NumberFormat from 'react-number-format';
+
+const dataTest = [
+    { name: 'March', Tested: 4000, Hospitalized: 2400, Dead: 2400 },
+    { name: 'April', Tested: 3000, Hospitalized: 1398, Dead: 2210 },
+    { name: 'May', Tested: 2000, Hospitalized: 9800, Dead: 2290 },
+    { name: 'June', Tested: 2780, Hospitalized: 3908, Dead: 2000 },
+    { name: 'July', Tested: 1890, Hospitalized: 4800, Dead: 2181 },
+    { name: 'August', Tested: 2390, Hospitalized: 3800, Dead: 2500 },
+    { name: 'September', Tested: 3490, Hospitalized: 4300, Dead: 2100 },
+];
+
+const DataRenderFunc =(props)=>{
+    const {stateId} = props
+    
+    const [loading, setLoading] = useState(true)
+    const [allStates, setAllStates] = useState(null)
+    const [showTesting, setShowTesting] = useState(true)
+    const [showHosp, setShowHosp] = useState(false)
+    const [showDeath, setShowDeath] = useState(false)
+     const [usaState, setUsaState] = useState(null)
+
+
+        
+    // const getData = async () => {
+    //     const url = "https://api.covidtracking.com/v1/states/current.json";
+    //     const response = await fetch(url);
+    //     const data = await response.json();
+    //     console.log(data);
+    //     return data;
+    // };
+
+    // Component fires when marker is clicked
+    useEffect(()=>{
+        const getData = async () => {
+            // const result = await axios(
+                const url = "https://api.covidtracking.com/v1/states/current.json";
+        const response = await fetch(url);
+        const data = await response.json();
+        console.log(data);
+        
+        setAllStates(data)
+         setUsaState(data[stateId])
+        setLoading(false)
+        setShowTesting(true)
+        }
+        getData()
+        // this.setState({ allStates: data, usaState: data[this.props.stateId], loading: false });
+    }, [])
+
+    //Toggle logic for testing data
+     const toggleTesting =()=>{
+        setShowTesting(!showTesting)
+    }
+
+    //Toggle logic for hospitalization data
+   const  toggleHosp=() =>{
+        setShowHosp(!showHosp)
+    }
+
+    //Toggle logic for death data
+    const toggleDeath=() => {
+        setShowDeath(!showDeath)
+    }
```

Comment:
Use functional form update instead https://reactjs.org/docs/hooks-reference.html#functional-updates

Reply:
I read the linked documentation section, yet still a bit unclear on how the functional update should be applied to the toggle functions. Do I link them to the useState for each of the toggles? What is the advantage to using functional form update as opposed to the current way it is set up? (Not disagreeing, genuinely curious and trying to learn the why behind it).

Reply:
https://sentry.io/answers/forget-set-state-is-async/#:~:text=The%20setState%20function%20takes%20an,after%20the%20state%20is%20changed.&text=This%20function%20will%20get%20called,updated%20value%20of%20the%20state.